BrowserKit/Sources/TabDataStore/TabDataStore.swift

```diff
@@ -96,20 +125,38 @@ public actor DefaultTabDataStore: TabDataStore {
     }
 
     // MARK: Saving Data
-    public func saveWindowData(window: WindowData) async {
+    public func saveWindowDataWithBackup(window: WindowData) async {
         Task {
-            if let windowSavingPath = self.windowURLPath(for: window.id) {
+            if let windowSavingPath = self.windowURLPath(for: window.id),
+               let backupWindowSavingPath = self.windowBackupURLPath(for: window.id) {
```

Comment:
I don't think this check belongs here. If we fail for some reason to generate a good path for the back up then it doesn't make sense to bail on attempting to save the main file anyway.